staging/src/k8s.io/client-go/examples/leader-election/main.go

```diff
@@ -135,11 +136,24 @@ func main() {
 			OnStartedLeading: func(ctx context.Context) {
 				// we're notified when we start - this is where you would
 				// usually put your code
+				startedLeading = true
```

Comment:
Do we need to worry about concurrency for access to `startedLeading`?

Reply:
`go run -race` doesn't flag it as a problem, but that's no proof that there is no race.

Reply:
Thank you for pointing this out. I have modified the code to use the atomic package to avoid this issue. From the code, I'm not entirely sure if there would be a data race problem (I think there isn't one). My thought was that, since this is an example, we need to keep it simple enough.

Reply:
I would argue the other way: because this is an example that folks will copy, it needs to be correct :grin: